backend/src/order.ts

```diff
@@ -72,88 +79,62 @@ export function checkExpired(): Promise<any> {
     {expire_minutes: config.expire_minutes});
 }
 
-export function checkPaytrailStatus(order_id: number): Promise<string> {
-  var verification = [config.paytrail.password, config.paytrail.user, PAYTRAIL_PREFIX + order_id].join('&');
-  var authcode = md5(verification).toUpperCase();
-
-  return new Promise<string>((resolve, reject) => {
-    var jar = request.jar();
-    request.post({
-        url: 'https://payment.paytrail.com/check-payment',
-        form: {
-          MERCHANT_ID: config.paytrail.user,
-          ORDER_NUMBER: PAYTRAIL_PREFIX + order_id,
-          AUTHCODE: authcode,
-          VERSION: '2',
-          CULTURE: 'fi_FI'
-        },
-        followAllRedirects: true, // paytrail sadness
-        jar: jar
-      },
-      function(err, response: any, body: string) {
-        (err) ? reject(err) : resolve(body);
-      });
-  }).then((body: string) => {
-    return new Promise<string>((resolve, reject) => {
-      jsdom.env({
-        html: body,
-        src: [fs.readFileSync('./node_modules/jquery/dist/jquery.js', 'utf-8')],
-        done: (err, window: any) => {
-          if (err) {
-            return reject(err);
-          }
-
-          var $ = window.$;
-          var rows = $('table tr');
-          if (rows.length !== 2) {
-            return reject('Row count doesnt match for order ' + order_id + ': got ' + rows.length + ' - expected only' +
-              ' headers ' + 'and one other');
-          }
-
-          var cells = $(rows[1]).find('td');
-          if (cells.length !== 4) {
-            return reject('Column count doesnt match for order ' + order_id + ': got ' + cells.length + ', expected ' +
-              'four');
-          }
-
-          var text = cells[3].innerHTML;
-          var possible_status = {
-            'Maksettu (Maksu vahvistettu)': 'paid',
-            'Maksettu (Suoritettu)': 'paid',
-            'Peruuntunut': 'cancelled',
-            'Peruuntunut (Ei maksettu)': 'cancelled',
-            'Odottaa maksua': 'pending-payment'
-          };
-
-          if (_.has(possible_status, text)) {
-            return resolve(possible_status[text]);
-          } else {
-            return reject('Unknown status for order ' + order_id + ': "' + text + '"');
-          }
-        }
-      });
-    });
-  });
+export async function checkPaymentStatus(order_id: number): Promise<string> {
+
+  let resp: IStatusResponse, err: Error, res: any, handler: IPayment;
+
+  log.log('debug', 'Check payment status', {order_id});
+
+  log.log('debug', 'Acquiring payment_id and payment_url', {order_id});
+  [res, err] = await to(db.query('select payment_id, payment_url from nk2_orders where id = :order_id', {order_id}));
```

Comment:
If you want to start using async/await (I'm not against, at least in C# it's quite awesome :p) let's use it properly. No need for the "to"-function - awaiting a promise throws if the promise rejects, so you can just wrap the whole function into a try-catch. Throwing from an async function returns a rejected promise, and returning a valid value returns a fulfilled promise, so AFAIK also the reject and resolve utils are unnecessary.

Reply:
Ok, I'll remove those resolve, reject -helpers. I'm using to because of golang style error handling, where error is always a value instead of something throwable/catchable. Take it as a syntatic sugar, because try-catch is always compiled to promise syntax

Reply:
Refactored to try/catch syntax